### DISCUSSION TITLE
MacOs support

### DISCUSSION THREAD

**Discussion by codeninja-ru** [mse-org/mseide-msegui#65] [General]
How difficult is it to add MacOs support? Do you think it will be possible in the near future? I believe it would be beneficial since many developers use Apple laptops for work, and it could help popularize MSEide+MSEgui

**Comment by fredvs** [comment]
Hello.
Of course it will be possible to add MacOS support!
But the major problem is: who has the skill and energy to do it?
Personally I don't have the level to do it. I don't have a Mac computer and every time I tried to install the Mac OS on a virtual computer I failed.

Msegui is an open-source project and everyone is welcome to improve it. For the future I think it would be good to have compatibility with Wayland too (even if msegui is yet perfectly compatible with XWayland).
There are some steps here: https://github.com/fredvs/wayland-pascal

But yes, MacOS support will be a great plus.

**Comment by fredvs** [comment]
>How difficult is it to add MacOs support?

Hum, I dont know well how Mac works, but you may take a look how LCL-Lazarus did for customdraw-for-mac (if there is one).
fpGUI has MacOS support and inspiration can come from it but it is no more useful because it uses [XQuartz](https://www.xquartz.org/) who is no more used in last MacOS.

**Comment by fredvs** [comment]
It is possible to have now MSEgui apps working on MacOS via XQuartz:

![mse_darwin](https://github.com/mse-org/mseide-msegui/assets/3421249/67aeef6f-c1f3-4bef-a60c-c7f1a745e6a0)


![strumpract_darwin](https://github.com/mse-org/mseide-msegui/assets/3421249/06d87cc6-40cd-4068-86f5-faa6ae76b0a4)

**Comment by fredvs** [comment]
Since commit: https://github.com/mse-org/mseide-msegui/commit/675bf9b8c20604e0cefc5e880e4264d75ab4af86

MSEide can compile and run project too on the Mac.
[EDIT] Also the Targetconsole is working.

![mseide_mac](https://github.com/mse-org/mseide-msegui/assets/3421249/d59a0f69-ed53-4d08-8adf-68b494eff574)

**Reply by fredvs** [reply]
Added binary of MSEide.app for MacOs in https://github.com/mse-org/mseide-msegui/releases/tag/5.10.0
